src/test/java/com/conveyal/gtfs/validator/FlexValidatorTest.java

```diff
@@ -100,192 +141,285 @@ private static Stream<LocationArguments> createLocationChecks() {
-    void validateStopTimeArrivalDepartureStartEndPickupDropoffTests(BaseArguments baseArguments) {
-        List<NewGTFSError> errors = FlexValidator.validateStopTime(
-            (StopTime) baseArguments.testObject,
-            null,
-            null
-        );
+    @MethodSource("createArrivalTimeTests")
+    void validateArrivalTime(BaseArguments baseArguments) {
+        List<NewGTFSError> errors = new ArrayList<>();
+        FlexValidator.validateArrivalTime((StopTime) baseArguments.testObject, errors);
         checkValidationErrorsMatchExpectedErrors(errors, baseArguments.expectedErrors);
     }
 
-    private static Stream<BaseArguments> createStopTimeChecksForArrivalDepartureStartEndPickupDropoff() {
+    private static Stream<BaseArguments> createArrivalTimeTests() {
         return Stream.of(
             new BaseArguments(
-                createStopTime(1130, INT_MISSING, INT_MISSING, INT_MISSING),null
+                createStopTimeForArrivalTimeTest(1130, INT_MISSING, INT_MISSING, INT_MISSING),null
             ),
             new BaseArguments(
-                createStopTime(INT_MISSING, 1330, INT_MISSING, INT_MISSING),null
+                createStopTimeForArrivalTimeTest(1130, INT_MISSING, 1200, INT_MISSING),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_ARRIVAL_TIME)
             ),
             new BaseArguments(
-                createStopTime(1130, 1330, INT_MISSING, INT_MISSING),null
+                createStopTimeForArrivalTimeTest(1130, INT_MISSING, INT_MISSING, 1200),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_ARRIVAL_TIME)
+            )
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("createDepartureTimeTests")
+    void validateDepartureTime(BaseArguments baseArguments) {
+        List<NewGTFSError> errors = new ArrayList<>();
+        FlexValidator.validateDepartureTime((StopTime) baseArguments.testObject, errors);
+        checkValidationErrorsMatchExpectedErrors(errors, baseArguments.expectedErrors);
+    }
+
+    private static Stream<BaseArguments> createDepartureTimeTests() {
+        return Stream.of(
+            new BaseArguments(
+                createStopTimeForDepartureTimeTest(1130, INT_MISSING, INT_MISSING, INT_MISSING),null
+            ),
+            new BaseArguments(
+                createStopTimeForDepartureTimeTest(1130, INT_MISSING, 1200, INT_MISSING),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_DEPARTURE_TIME)
             ),
             new BaseArguments(
-                createStopTime(INT_MISSING, INT_MISSING, 1100, INT_MISSING),null
+                createStopTimeForDepartureTimeTest(1130, INT_MISSING, INT_MISSING, 1200),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_DEPARTURE_TIME)
+            )
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("createStopIdTests")
+    void validateStopId(BaseArguments baseArguments) {
```

Comment:
You could spread `baseArguments` and put its attributes as this method's arguments instead.